WikiPreview.Fluent.Plugin/WikiPreviewSearchApp.cs

```diff
@@ -108,59 +109,88 @@ public ValueTask LoadSearchApplicationAsync()
-                            string imgUrl = value.Thumbnail.Source;
-                            using var imageClient = new HttpClient();
-                            Stream stream = await imageClient.GetStreamAsync(imgUrl, cancellationToken)
+                            WikiPreviewSearchResult wikiPreviewSearchResult =
+                                await GenerateSearchResult(entry.Value, searchedText).ConfigureAwait(false);
+                            await channel.Writer.WriteAsync(wikiPreviewSearchResult, CancellationToken.None)
                                 .ConfigureAwait(false);
-                            bitmapImageResult = new BitmapImageResult(new Bitmap(stream));
-                        }
-                        else
-                        {
-                            bitmapImageResult = new BitmapImageResult(); // create empty if no image source.
-                        }
-
-                        WikiPreviewSearchResult wikiPreviewSearchResult = new()
-                        {
-                            Url = wikiUrl,
-                            PreviewImage = bitmapImageResult,
-                            DisplayedName = displayedName,
-                            ResultName = resultName,
-                            SearchedText = searchedText,
-                            Score = score,
-                            PageId = pageId,
-                            AdditionalInformation = additionalInfo
-                        };
-                        await channel.Writer.WriteAsync(wikiPreviewSearchResult).ConfigureAwait(false);
-                    }, cancellationToken).ContinueWith(_ => channel.Writer.Complete());
+                        }, cancellationToken)
+                        .ContinueWith(_ => channel.Writer.Complete(), CancellationToken.None);
                 else
                     channel.Writer.Complete();
-            });
+            }, CancellationToken.None);
 
             await foreach (WikiPreviewSearchResult item in channel.Reader.ReadAllAsync(cancellationToken))
                 yield return item;
         }
 
-        public ValueTask<ISearchResult> GetSearchResultForId(string serializedSearchObjectId)
+        public ValueTask<WikiPreviewSearchResult> GetSearchResultForId(string serializedSearchObjectId)
+        {
+            try
+            {
+                if (string.IsNullOrWhiteSpace(serializedSearchObjectId))
+                    return default;
+
+                string url = "https://en.wikipedia.org/w/api.php?action=query&prop=extracts|pageimages&pageids=" +
+                             serializedSearchObjectId +
+                             "&explaintext&exintro&pilicense=any&pithumbsize=100&format=json";
+
+                using var httpClient = new HttpClient();
+                Task<Wiki> task = httpClient.GetFromJsonAsync<Wiki>(url, _serializerOptions);
+                task.Wait();
+                PageView pageView = task.Result.Query.Pages.Values.First();
```

Comment:
You should check here whether `Values` is empty before calling `First`, catching exceptions is relatively heavy and this can cause a little spike while searching.

Reply:
Checking for the empty using `Count`.